src/Adaptor.js

```diff
@@ -265,21 +268,35 @@ export function upsert(table, uuid, record) {
         ON CONFLICT ${conflict}
         DO UPDATE SET ${updateValues};`;
 
-      return new Promise((resolve, reject) => {
-        console.log(`Executing upsert via : ${safeQuery}`);
-
-        client.query(query, (err, result) => {
-          if (err) {
-            reject(err);
-            client.end();
-          } else {
-            console.log(result);
-            resolve(result);
-          }
+      execute = () => {
+        return new Promise((resolve, reject) => {
+          console.log(`Executing upsert via : ${safeQuery}`);
+
+          client.query(query, (err, result) => {
```

Comment:
At this point, @lakhassane , aren't you already executing the sql against the DB?